### ISSUE TITLE
[Feature Request] Store data to Cloud Storage

### ISSUE THREAD

**Issue by sagelga** [frzyc/genshin-optimizer#79]
Hi, I am currently using the site and got some inconvenient on the database.
I need to manually store the `data.json` file to Google Drive in order to make a database backup and allows me to edit the database on the other device.

Can you make the "Store in Google Drive" or any equivalent solution? 

Example website that already implements this feature is https://paimon.moe/. They use [Application Data Directory](https://developers.google.com/drive/api/v3/appdata) from Google Drive.

If the website allows that, it will helps me (and many others) to sync the database across their device. Pretty much a QOL improvement. 

Thanks!

**Comment by GabenRulez** [comment]
I fully agree. A function like that would be especially useful for people working on multiple devices.

**Comment by hyoretsu** [comment]
This. I just cleared my browser data and my database got wiped... At least I had a database backup from a month ago, but most people wouldn't have one.

**Comment by lampham7** [comment]
How is this feature request not high priority?

**Comment by kudoku** [comment]
+1 for this feature. I took a peek at the backup json and I wouldn't want this in my browser local storage long term as I have thousands of entries.

**Comment by isaswa** [comment]
+1

**Comment by Skynetexe** [comment]
+1 for this. This is the 3rd time so far that I've lost all of my data as well as my multi target optimizations. If for nothing else, this is incredible as a backup, and it would alleviate my worries that it will all just go poof without notice

**Comment by nguyentvan7** [comment]
In the meantime, you can backup your DB from the settings tab in the top right

The data is stored in localstorage, so it will only be wiped if the user deletes the cookies for the site, or uses some cleaning program like CCleaner.

The exception is any browser on iOS. Due to a privacy policy on iOS, browser cookies will be deleted after 2 weeks if the site has not been opened, so we recommend that iOS users regularly back up their DB until we have the availability to implement this feature

**Comment by nmasnadithya** [comment]
I would like to make an attempt at this, in my free time. I should just submit a PR or is there anything else, that I should be aware of?

**Comment by nguyentvan7** [comment]
Only thing to be aware of is that we deploy using Github pages and do not have any backend. So the API key will be fully public once the site is deployed. But AFAIK, Google allows you to scope the key to only work on a specific site, so shouldn't really be an issue.

**Comment by nambn** [comment]
anything i can help to speed up the deployment of this issue?

**Comment by nmasnadithya** [comment]
Life has been hectic, so I actually couldn't start working on this. So, if you are taking this up, you can go ahead.

**Comment by frzyc** [comment]
There are some additional UI work on that branch, as well as getting the site approved by Google (which has quite a few steps). We've not had much free time to address either of them so far.

**Comment by nguyentvan7** [comment]
with the addition of teams/loadouts, which are not easily recreated, this has even more reason to exist. Although we are working on our own backend, that still seems to be a ways away, and this seems like a weekend-sized feature that would provide a lot of benefit to our user base

**Comment by MRDGH2821** [comment]
Bumping this up!

Can we have other possible cloud backup options if not google drive?
